core/src/main/java/org/mskcc/cbio/oncokb/validation/validation.java

```diff
@@ -184,38 +214,189 @@ public static void main(String[] args) throws IOException, GeneralSecurityExcept
+                Set<String> articles = EvidenceUtils.getPmids(Collections.singleton(evidence));
+                setValue(row, "PMIDs", org.apache.commons.lang3.StringUtils.join(articles, ", "));
+                setValue(row, "NumberOfPMIDs", Integer.toString(articles.size()));
+
+                Set<ArticleAbstract> articleAbstracts = EvidenceUtils.getAbstracts(Collections.singleton(evidence));
+                Set<String> abstractContent = new HashSet<>();
+                for (ArticleAbstract aa : articleAbstracts) {
+                    abstractContent.add(aa.getAbstractContent());
+                }
+                setValue(row, "Abstracts", org.apache.commons.lang3.StringUtils.join(abstractContent, ", "));
+
+                service.insert(feedUrl, row);
+            }
+        }
+    }
+
+    private static void printTumorTypeSummary(Set<Evidence> evidences, SpreadsheetService service, WorksheetEntry entry) throws IOException, ServiceException {
+        URL feedUrl = entry.getListFeedUrl();
+        if (evidences != null && service != null && entry != null) {
+            for (Evidence evidence : evidences) {
+                ListEntry row = new ListEntry();
+                setValue(row, "Gene", evidence.getGene().getHugoSymbol());
+
+                List<String> alterationNames = new ArrayList<>();
+                for (Alteration alteration : evidence.getAlterations()) {
+                    if (StringUtils.isNullOrEmpty(alteration.getName())) {
+                        alterationNames.add(alteration.getAlteration());
+                    } else {
+                        alterationNames.add(alteration.getName());
+                    }
+                }
+                setValue(row, "Variants", MainUtils.listToString(alterationNames, ", "));
+
+                setValue(row, "CancerType", getCancerType(evidence.getOncoTreeType()));
+                setValue(row, "Summary", evidence.getDescription());
+                service.insert(feedUrl, row);
+            }
+        }
+    }
+
+    private static Set<Evidence> getPublishedEvidencesByLevel(LevelOfEvidence levelOfEvidence) throws IOException {
+        String json = FileUtils.readRemote("http://oncokb.org/legacy-api/evidence.json?levels=" + levelOfEvidence.name());
+
+        ObjectMapper mapper = new ObjectMapper();
+        List<List<Evidence>> list = mapper.readValue(json,
+            new TypeReference<ArrayList<ArrayList<Evidence>>>() {
+            });
+        return new HashSet<>(list.get(0));
+    }
+
+    private static void setValue(ListEntry row, String element, String value) {
+        if (value == null)
+            value = "";
+        row.getCustomElements().setValueLocal(element, value);
+    }
+
+    private static String getCancerType(TumorType tumorType) {
+        String cancerTypeName = "";
+        if (tumorType != null) {
+            if (StringUtils.isNullOrEmpty(tumorType.getName())) {
+                if (tumorType.getMainType() != null) {
```

Comment:
For cancer type name, I understood as we use SubType first. If SubType doesn't exist, then we use MainType. It is not the case here?